pkg/operator/controller/dns/controller.go

```diff
@@ -130,33 +141,32 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 	var statuses []iov1.DNSZoneStatus
 	for i := range zones {
 		zone := zones[i]
-		var conditions []iov1.DNSZoneCondition
-		err := r.dnsProvider.Ensure(record, zone)
-		if err != nil {
-			conditions = append(conditions, iov1.DNSZoneCondition{
-				Type:               iov1.DNSRecordFailedConditionType,
-				Status:             string(operatorv1.ConditionTrue),
-				Reason:             "ProviderError",
-				Message:            fmt.Sprintf("The DNS provider failed to ensure the record: %v", err),
-				LastTransitionTime: metav1.Now(),
-			})
+		condition := iov1.DNSZoneCondition{
+			Status:             string(operatorv1.ConditionUnknown),
+			Type:               iov1.DNSRecordFailedConditionType,
+			LastTransitionTime: metav1.Now(),
+		}
+
+		if len(zone.ID) == 0 {
+			condition.Status = string(operatorv1.ConditionTrue)
+			condition.Reason = "EmptyZoneId"
+			condition.Message = "Zone has an empty id"
+		} else if err := r.dnsProvider.Ensure(record, zone); err != nil {
+			condition.Status = string(operatorv1.ConditionTrue)
+			condition.Reason = "ProviderError"
+			condition.Message = fmt.Sprintf("The DNS provider failed to ensure the record: %v", err)
 			result.RequeueAfter = 30 * time.Second
+		} else {
+			condition.Status = string(operatorv1.ConditionFalse)
+			condition.Reason = "ProviderSuccess"
+			condition.Message = fmt.Sprintf("The DNS provider succeeded to ensure the record: %v", err)
```

Comment:
`err` is nil here, so omit it from the message.  Also, please change "succeeded to ensure" to "succeeded in ensuring".